app/controllers/QuestionnaireController.scala

```diff
@@ -16,110 +16,146 @@
 
 package controllers
 
-import _root_.forms.{ QuestionnaireDiversityInfoForm, QuestionnaireEducationInfoForm, QuestionnaireOccupationInfoForm }
-import config.CSRHttp
+import _root_.forms.{QuestionnaireDiversityInfoForm, QuestionnaireEducationInfoForm, QuestionnaireOccupationInfoForm}
 import connectors.ApplicationClient
 import connectors.ExchangeObjects.Questionnaire
 import models.CachedDataWithApp
-import play.api.mvc.{ Request, Result }
-import security.Roles.{ DiversityQuestionnaireRole, EducationQuestionnaireRole, OccupationQuestionnaireRole, StartQuestionnaireRole }
+import play.api.mvc.{Request, Result}
+import security.QuestionnaireRoles._
 import uk.gov.hmrc.play.http.HeaderCarrier
+import helpers.NotificationType._
 
 import scala.concurrent.Future
 import scala.language.reflectiveCalls
 
-object QuestionnaireController extends QuestionnaireController(ApplicationClient) {
-  val http = CSRHttp
-}
+object QuestionnaireController extends QuestionnaireController(ApplicationClient)
+
+class QuestionnaireController(applicationClient: ApplicationClient) extends BaseController(applicationClient) {
```

Comment:
I don't see any need of declaring abstract class here because class will let you create a test by simply instantiating and injecting the mock. Otherwise I will have to create a class that extends this controller for testing.